### ISSUE TITLE
empty plot when plotting with 'regionplot'

### ISSUE THREAD

**Issue by zhana-optima** [totajuliusd/topr#88]
Hi, when using regionplot, the desired plot appears in second place after an empty plot. In Rmd,e.g. 

![image](https://github.com/user-attachments/assets/b77dcfa4-aa00-461a-8600-1caab0247ac7)

**Comment by totajuliusd** [comment]
Hi, can you show me the command you are using to create the plot?

**Comment by zhana-optima** [comment]
Hello,

Here it is:    topr::regionplot(to_plot_list,
                                   region = rrid,
                                   ntop = length(to_plot_list),
                                   legend_labels = plot_legends,
                                   annotate_with = 'ID',
                                   build = 38,
                                   log_trans_p = !transform_log10,
                                   verbose = FALSE,
                                   rsids_with_vline = fit_snps)
________________________________

**Comment by zhana-optima** [comment]
Also, when I add the option for rsids_with_vline = rsids, I do get an issue with the legend labels which was previously reported and resolved.[cid:824602f7-24ed-4668-a6c8-0b2e5c892b41]
If I don't add the rsids_with_vline option, I get a normal legend:
[cid:39dcba67-9006-4d83-8f4a-ff370cbf4233]
________________________________

**Comment by totajuliusd** [comment]
Which editor are you using? I cant replicate the problem when I run the command from Rstudio or directly from the terminal. 

And which version of topr are you using?

**Comment by zhana-optima** [comment]
I am working in Rstudio on a Mac machine. I am using version 2.0.2 of topr.

Best,
Zhana
________________________________

**Comment by totajuliusd** [comment]
Does everything work as expected when you do:

`manhattan(CD_UKBB)`

and:

`ggplot()+geom_point(aes(x=c(1:10), y=c(1:10)))`

**Comment by zhana-optima** [comment]
Yes, both seem to be running fine.

________________________________

**Comment by totajuliusd** [comment]
Hmmm.... what about when you do the following?

` regionplot(CD_UKBB, gene="IL23R")`

**Comment by zhana-optima** [comment]
Yes, this looks good but there is no legend.
________________________________

**Comment by totajuliusd** [comment]
What if you do:

` regionplot(list(CD_UKBB, lead_snps), gene="IL23R", ntop=2, legend_labels=c("CD UKBB", "lead snps"), build = "38", rsids_with_vline = "rs11576518")`
 
 Although, I can see the issue there with the legend label you mentioned. That should have been fixed, but is clearly still a problem in the regionplot.

**Comment by zhana-optima** [comment]
Yes, with this example I do get both the empty plot first + the legend issue as well.


________________________________

**Comment by zhana-optima** [comment]
First plot
[cid:eaff7929-7ec9-44d7-8b9b-e265e374555f]
And then second plot
[cid:eef615d0-115f-4379-8264-e02e040b7645]
________________________________

**Comment by totajuliusd** [comment]
Which version of RStudio are you using?

And do you think its the inclusion of the legend that causes the empty plot to appear? For examples, e.g. is this ok

` regionplot(CD_UKBB, gene="IL23R")`
 
 but not this:
 
`  regionplot(CD_UKBB, gene="IL23R", legend_labels='CD_UKBB')`

**Comment by zhana-optima** [comment]
This is my Rstudio version: Version 2023.12.1+402 (2023.12.1+402)
The usage of regionplot triggers an empty plot regardless of the arguments.
________________________________

**Comment by totajuliusd** [comment]
Does the following also create an empty plot?

```
jpeg("regionplot.jpg")
regionplot(CD_UKBB, gene="IL23R")
dev.off()
```

I cant seem to re-create this problem. Im using RStudio version 2024.04.2+764 (2024.04.2+764) and R R version 4.4.0, and ggplot2 version 3.5.1.

**Comment by zhana-optima** [comment]
No, it does not produce an empty plot. I will update my Rstudio version and keep you posted if the problem still persists. It could be due to versions...
________________________________

**Comment by zhana-optima** [comment]
I have updated both R and Rstudio and both issues (empty plot + legend names) still persist. If I use extract_plots = TRUE, there is no issue, and each sub-plot appears on its own. It is when this defaults to FALSE and the three plots are combined the empty plot issue appears.

Best,
Zhana
________________________________

**Comment by RvV1979** [comment]
I can confirm that I have the same issues of an empty first page when exporting to PDF. I am running Rstudio 2024.04.2+764  on an Apple M1 Pro. 
It is not a major issue, however, because I can easily remove the blank page in post.
